src/index.tsx

```diff
@@ -13,72 +12,66 @@ interface AccorionProps {
   [rest: string | number | symbol]: unknown;
 }
 
-let interval: number;
+interface ListItemProps {
+  SummaryComponent: React.ElementType;
+  DetailComponent: React.ElementType;
+  id: string | number;
+}
 
-const Accordion = ({ items, ...rest }: AccorionProps) => {
-  const [opened, setOpened] = useState<Record<string, boolean>>({});
+const ListItem = ({
+  id,
+  SummaryComponent,
+  DetailComponent,
+  ...rest
+}: ListItemProps) => {
+  const contentItem = useRef<HTMLDivElement>(null);
+  const [opened, setOpened] = useState<boolean>(false);
```

Comment:
Also this will create a new state in every LI so I am not sure about the memory usage.

Reply:
The memory usage actually seems lower in my initial tests. I'll post memory snapshot data.

Reply:
Changes in the PR memory - https://imgur.com/a/J1ieNTs
Current memory - https://imgur.com/a/BqlC0NI